presto-tpcds/src/main/java/com/facebook/presto/tpcds/statistics/TpcdsTableStatisticsFactory.java

```diff
@@ -43,25 +56,50 @@ private TableStatistics toTableStatistics(Map<String, ColumnHandle> columnHandle
         if (rowCount > 0) {
             Map<String, ColumnStatisticsData> columnsData = statisticsData.getColumns();
             for (Map.Entry<String, ColumnHandle> entry : columnHandles.entrySet()) {
-                tableStatistics.setColumnStatistics(entry.getValue(), toColumnStatistics(columnsData.get(entry.getKey()), rowCount));
+                TpcdsColumnHandle columnHandle = (TpcdsColumnHandle) entry.getValue();
+                tableStatistics.setColumnStatistics(entry.getValue(), toColumnStatistics(columnsData.get(entry.getKey()), columnHandle.getType(), rowCount));
             }
         }
 
         return tableStatistics.build();
     }
 
-    private ColumnStatistics toColumnStatistics(ColumnStatisticsData columnStatisticsData, long rowCount)
+    private ColumnStatistics toColumnStatistics(ColumnStatisticsData columnStatisticsData, Type type, long rowCount)
     {
         ColumnStatistics.Builder columnStatistics = ColumnStatistics.builder();
         long nullCount = columnStatisticsData.getNullsCount();
         columnStatistics.setNullsFraction(new Estimate(nullCount / rowCount));
         columnStatistics.addRange(builder -> builder
-                .setLowValue(columnStatisticsData.getMin())
-                .setHighValue(columnStatisticsData.getMax())
+                .setLowValue(
+                        columnStatisticsData.getMin()
+                                .map(value -> toPrestoValue(value, type)))
+                .setHighValue(
+                        columnStatisticsData.getMax()
+                                .map(value -> toPrestoValue(value, type)))
                 .setDistinctValuesCount(new Estimate(columnStatisticsData.getDistinctValuesCount()))
                 .setFraction(new Estimate((rowCount - nullCount) / rowCount))
                 .build());
 
         return columnStatistics.build();
     }
+
+    private Object toPrestoValue(Object tpcdsValue, Type type)
+    {
+        if (type instanceof VarcharType) {
+            return Slices.utf8Slice((String) tpcdsValue);
+        }
+        else if (type instanceof CharType) {
+            return trimSpacesAndTruncateToLength(Slices.utf8Slice((String) tpcdsValue), type);
+        }
+        else if (type.equals(BIGINT) || type.equals(INTEGER) || type.equals(DATE) || type instanceof DecimalType) {
```

Comment:
s/`type instanceof DecimalType`/`(type instanceof DecimalType and isShortDecimal....)`